cloudinit/util.py

```diff
@@ -625,10 +630,9 @@ def get_linux_distro():
                     flavor = match.groupdict()['codename']
         if distro_name == 'rhel':
             distro_name = 'redhat'
-    elif os.path.exists('/bin/freebsd-version'):
-        distro_name = 'freebsd'
-        distro_version, _ = subp(['uname', '-r'])
-        distro_version = distro_version.strip()
+    elif platform.system():
```

Comment:
hrm, I think we may need to move this inside the else block below, and after we try platform.dist() because most linux systems will report only 'Linux' on this elif clause, but we want to rely on platform.dist() if supported, before leveraging the simplistic response from platform.system(). Since you may want *bsd to source platform.system and platform.release instead of platform.dist, I suggest we do either a file exists test for *BSE or `system_info()['variant'] in ("freebsd", "netbsd")` or use the `is_NetBSD` `is_FreeBSD` checks